### PR TITLE
Theme Showcase: Move "Install theme" button into header area to be more similar to "Install plugin" button from plugins page.

### LINKED ISSUE #52521: Theme Showcase: Move "Install theme" button into header area
We made a point of moving the primary action for installing plugins to the header like so:

Screen Shot 2021-05-03 at 1 04 55 PM

My Home and Support pages follow a similar pattern, so we should do the same for Themes (`/themes`):

Screen Shot 2021-05-03 at 10 19 25 AM

This also removes the weird dead space between the currently active theme bar and the list of themes.

### PR DESCRIPTION
**Pull request by DanielSmithMichigan**
#### Changes proposed in this Pull Request

* Move the 'Install theme' button above the 'Current Theme' section so that the layout of the page is more analogous to the 'Install plugin' button

[Before Desktop](https://user-images.githubusercontent.com/10274366/118141364-48768700-b3d7-11eb-8862-b2e4cc703669.png)

[Before Mobile](https://user-images.githubusercontent.com/10274366/118141291-3268c680-b3d7-11eb-92ea-aa6d78a3afdb.png)

[After Desktop](https://user-images.githubusercontent.com/10274366/118141057-fa618380-b3d6-11eb-8423-24283cddd999.png)

[After Mobile](https://user-images.githubusercontent.com/10274366/118141128-106f4400-b3d7-11eb-9d69-66b83da6bf54.png)

[Plugins Page For Reference](https://user-images.githubusercontent.com/10274366/117474846-5b95dc80-af29-11eb-9791-058a2cdb9e8f.png)

#### Testing instructions

Previously existing behavior from trunk for install theme button should be maintained, for example:
- For any view which used to have an install theme button on trunk, it should still be there on my branch.
- For any view which did NOT used to have an install theme button on trunk, it should NOT be there on my branch.
- Just as it works on trunk, when a site is selected, install theme button should link to: /themes/upload/SITE_SLUG
- Just as it works on trunk, when a site is NOT selected, install theme button should link to: /themes/upload. Clicking it will prompt you to select a site.
- Clicking the install theme button should record an `upload theme` Google analytics event
- Clicking the install theme button should record a `calypso_click_theme_upload` tracks event
- The "Install theme" text on the button should translate to other languages
- Test different browsers, screen sizes and devices at your own discretion

Views to consider:
- /themes/SITE_SLUG page for a simple site
- /themes/SITE_SLUG page for an atomic site
- /themes/SITE_SLUG page for a self-hosted multisite install connected via jetpack
- /themes page rendered via Server-side rendering
- /themes page while logged out
- /themes page while logged in (/themes page for "All my sites")

Related to #52521

### CODE REVIEW

**Comment by DanielSmithMichigan** [comment]
@sixhours Currently there's still a difference between the plugins and themes install buttons: the install themes button is a 'compact' button, and the install plugins button is not.

What are your thoughts about using the non-compact view for "themes?

I think they look about the same on desktop, but NON-compact looks much better on mobile.

[Desktop, NOT compact](https://user-images.githubusercontent.com/10274366/117482975-24c4c400-af33-11eb-97c3-9f232cb6f6ed.png)

[Mobile, NOT compact](https://user-images.githubusercontent.com/10274366/117482935-17a7d500-af33-11eb-917b-2ab0f3f5d505.png)

[Desktop, compact](https://user-images.githubusercontent.com/10274366/117483042-373efd80-af33-11eb-8049-bc253726742a.png)

[Mobile, compact](https://user-images.githubusercontent.com/10274366/117483097-4625b000-af33-11eb-8028-aa875bb0aa45.png)


For reference:

[Plugins page, desktop](https://user-images.githubusercontent.com/10274366/117483194-6eadaa00-af33-11eb-840c-ae917af7ae08.png)

[Plugins page, mobile](https://user-images.githubusercontent.com/10274366/117483246-81c07a00-af33-11eb-8634-07ebc22dbd20.png)

**Comment by sixhours** [comment]
> What are your thoughts about using the non-compact view for "themes?

I definitely think we should use the non-compact version so it's better aligned with Plugins (and other pages like My Home or Support)!

**Comment by mreishus** [comment]
You might be able to simplify by defining the handler function inline to the component:

```diff
diff --git a/client/my-sites/themes/install-theme-button.jsx b/client/my-sites/themes/install-theme-button.jsx
index e8275a21c5..ae89f2aef4 100644
--- a/client/my-sites/themes/install-theme-button.jsx
+++ b/client/my-sites/themes/install-theme-button.jsx
@@ -29,14 +29,6 @@ import Gridicon from 'calypso/components/gridicon';
  */
 import './install-theme-button.scss';
 
-function InstallThemeButtonOnClick( { trackUploadClick, trackATUploadClick, ATEnabled } ) {
-       trackClick( 'upload theme' );
-       trackUploadClick();
-       if ( ATEnabled ) {
-               trackATUploadClick();
-       }
-}
-
 const InstallThemeButton = connectOptions(
        ( { isMultisite, isLoggedIn, trackUploadClick, trackATUploadClick, ATEnabled, siteSlug } ) => {
                // I'm not sure if this below code is needed, or if we can enforce these conditions by only including this code in the appropriate pages.
@@ -44,16 +36,18 @@ const InstallThemeButton = connectOptions(
                        return null;
                }
 
+               const installThemeButtonOnClick = () => {
+                       trackClick( 'upload theme' );
+                       trackUploadClick();
+                       if ( ATEnabled ) {
+                               trackATUploadClick();
+                       }
+               };
+
                return (
                        <Button
                                className="themes__upload-button"
-                               onClick={ () =>
-                                       InstallThemeButtonOnClick( {
-                                               trackUploadClick,
-                                               trackATUploadClick,
-                                               ATEnabled,
-                                       } )
-                               }
+                               onClick={ installThemeButtonOnClick }
                                href={ siteSlug ? `/themes/upload/${ siteSlug }` : '/themes/upload' }
                        >
                                <Gridicon icon="cloud-upload" className="themes__upload-button-gridicon" />
```
(untested)

It's true that the function would be recreated each render, but I don't think it would make any sort of actual performance impact. It could also be sidestepped with useCallback but that's [probably not needed](https://dmitripavlutin.com/dont-overuse-react-usecallback/).

I'm also fine w/ the current implementation.

**Comment by mreishus** [comment]
We discovered the `onClick` handler is not working properly in development on multisite, however, the problem is fixed when SSR is disabled by removing `isometric` in `sections.js`.
Multisite is also slated to be removed in #52686

**Comment by sixhours** [comment]
Gave this a spin and it's looking good! I would remove the icon from the button entirely, since we don't use icons elsewhere in this header area. I also noticed the Tracks event is no longer firing but I suspect this is related to the `onClick` handler not working.

**Comment by DanielSmithMichigan** [comment]
> Gave this a spin and it's looking good! I would remove the icon from the button entirely, since we don't use icons elsewhere in this header area. I also noticed the Tracks event is no longer firing but I suspect this is related to the `onClick` handler not working.

Did you find that the tracks event wasn't firing in the **multisite** context? Or were there other situations the track event was not firing?

**Comment by sixhours** [comment]
> Did you find that the tracks event wasn't firing in the multisite context? Or were there other situations the track event was not firing?

Hmm, is the multisite context an account with multiple sites? If so, correct, the Tracks event was not registered on my account with multiple sites. :D

**Comment by DanielSmithMichigan** [comment]
> Hmm, is the multisite context an account with multiple sites? If so, correct, the Tracks event was not registered on my account with multiple sites. :D

I think I'm caught up using terminology I was reading in the code without explaining it 🙂.

This is the behavior we were observing:

We don't expect the Tracks event to be recorded when: Clicking "install theme" from the /themes page [after selecting All my sites](https://user-images.githubusercontent.com/10274366/118145153-4b737680-b3db-11eb-851f-c2a067a152cc.png)

We do expect the Tracks event to be recorded when: Clicking "install theme" from the /themes page [after selecting a specific site](https://user-images.githubusercontent.com/10274366/118144887-f8012880-b3da-11eb-94ee-f052385b0d3f.png)

**Comment by sixhours** [comment]
> We do expect the Tracks event to be recorded when: Clicking "install theme" from the /themes page after selecting a specific site

Aha, thank you for that clarification! This was the case for me; I was under one of my sites (not all sites) and I don't see a Tracks event when I click "Install theme". 

I'm logging the analytics events to local storage with this line in my browser console:

`localStorage.debug='calypso:analytics*'`

Which shows an output like this upon clicking "Install theme" in a single site context:

Screen Shot 2021-05-13 at 11 17 15 AM

I do see the Google Analytics event but nothing for Tracks. I haven't fiddled with the code to test it, but I suspect `recordTracksEvent` needs to mapped to dispatch:

Old method using `mapDispatchToProps`:

https://github.com/Automattic/wp-calypso/pull/52671/files#diff-5e6c175ac60d9ed48935454e0bcf1861c5d5893c33a08edb946162d2e0dea550L447

New method, no dispatch:

https://github.com/Automattic/wp-calypso/pull/52671/files#diff-944bf6e20fa7aa994372f5869f6e0d4e08ea825af43968aadc28a6e4d7628f8aR33

**Comment by DanielSmithMichigan** [comment]
> > We do expect the Tracks event to be recorded when: Clicking "install theme" from the /themes page after selecting a specific site
> 
> Aha, thank you for that clarification! This was the case for me; I was under one of my sites (not all sites) and I don't see a Tracks event when I click "Install theme".
> 
> I'm logging the analytics events to local storage with this line in my browser console:
> 
> `localStorage.debug='calypso:analytics*'`
> 
> Which shows an output like this upon clicking "Install theme" in a single site context:
> 
> Screen Shot 2021-05-13 at 11 17 15 AM
> 
> I do see the Google Analytics event but nothing for Tracks. I haven't fiddled with the code to test it, but I suspect `recordTracksEvent` needs to mapped to dispatch:
> 
> Old method using `mapDispatchToProps`:
> 
> https://github.com/Automattic/wp-calypso/pull/52671/files#diff-5e6c175ac60d9ed48935454e0bcf1861c5d5893c33a08edb946162d2e0dea550L447
> 
> New method, no dispatch:
> 
> https://github.com/Automattic/wp-calypso/pull/52671/files#diff-944bf6e20fa7aa994372f5869f6e0d4e08ea825af43968aadc28a6e4d7628f8aR33

Ok, interesting.

That was **not** the behavior I was expecting. I will check that out, thank you!

**Comment by DanielSmithMichigan** [comment]
@sixhours Thank you for finding that issue; it was a real issue, and it was resolved like your suggestion.

I didn't know about the shorthand JSON form for redux mapDispatchToProps, so I didn't notice that action was being dispatched. But the testing instructions were helpful and that issue is resolved now.

**Comment by DanielSmithMichigan** [comment]
I resolved some merge conflicts on this PR this morning, related to [this other pull request](https://github.com/Automattic/wp-calypso/pull/52776).

I did another review and testing pass over. At this point, I'm just waiting on a PR review approval! (I think)

File: client/my-sites/themes/themes-header.scss
Comment (mreishus) [inline client/my-sites/themes/themes-header.scss]: I believe this rule does nothing because `.themes_header` is not a flex child.

**Comment by mreishus** [comment]
Tested all environments from a visual standpoint and they look good: 

- :heavy_check_mark: multisite 
- :heavy_check_mark: Simple site 
- :heavy_check_mark: Atomic site
- :heavy_check_mark: Jetpack site [A regular jetpack installation was sufficient for me to test client/my-sites/themes/single-site-jetpack.jsx]
- :heavy_check_mark: /themes logged out
- :heavy_check_mark: /themes logged out, JS off

Recommend removing the CSS rule as above.

Just need someone else to approve the tracks events and we're g2g. (Maybe that'll be me if I can learn more about it, haven't tested tracks in a while)

File: client/my-sites/themes/themes-header.scss
Comment (DanielSmithMichigan) [reply client/my-sites/themes/themes-header.scss]: This makes sense, I removed the property and retested.

**Comment by sixhours** [comment]
The Tracks event is working properly for me now! 🚢

**Review by mreishus — APPROVED** [review]
LGTM, I'm seeing the tracks events when ? `localStorage.debug='calypso:analytics*'` and I saw at least one in the live tracks viewer.

**Comment by DanielSmithMichigan** [comment]
Created [this followup](https://github.com/Automattic/wp-calypso/issues/52927) to address the unused code which was removed.

**Comment by DanielSmithMichigan** [comment]
@mreishus How should the [related documentation](https://wordpress.com/support/themes/uploading-setting-up-custom-themes/#upload-the-theme) be addressed? Is it safe to merge before that's been updated?

**Comment by mreishus** [comment]
Discussed in slack, and I believe this is safe to merge before it's been updated; it is a relatively minor change.

**Pull request merged**
